File: backend/src/data_models.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

class ProjectBasic(BaseModel):
    id: str
    name: str

class Project(ProjectBasic):
    description: Optional[str] = None

class LocationBasic(BaseModel):
    id: str
    name: str

class Location(LocationBasic):
    description: Optional[str] = None

class TagBasic(BaseModel):
    id: str
    name: str

class Tag(TagBasic):
    description: Optional[str] = None

class TaskBasic(BaseModel):
    id: str
    name: str
    description: Optional[str] = None

class Task(TaskBasic):
    project_id: Optional[str] = None
    location_id: Optional[str] = None
    is_done: Optional[str] = None
    tags: Optional[list[str]] = None # tag ids

File: ios/TaskTracker/TaskTracker/DataModels.swift
//
//  DataModels.swift
//  TaskTracker
//
//  Created by Aleksandrs Maklakovs on 17/08/2024.
//

import Foundation

struct TaskBasic: Codable, Hashable {
    var id: String = ""
    var name: String = ""
    var is_done: Bool = false
}

struct Task: Codable, Hashable {
    var id: String = ""
    var name: String = ""
    var description: String = ""
    var project_name: String = ""
    var location_name: [String] = []
    var is_done: Bool = false
    var tags: [String] = []
}

struct ProjectBasic: Codable, Hashable {
    var id: String = ""
    var name: String = ""
}

struct Project: Codable, Hashable {
    var id: String = ""
    var name: String = ""
    var description: String = ""
}

struct LocationBasic: Codable, Hashable {
    var id: String
    var name: String
}

struct Location: Codable, Hashable {
    var id: String
    var name: String
    var description: String
}

struct TagBasic: Codable, Hashable {
    var id: String
    var name: String
}

struct Tag: Codable, Hashable {
    var id: String
    var name: String
    var description: String
}
